src/eva/vision/data/datasets/classification/panda.py

```diff
@@ -0,0 +1,160 @@
+"""Dataset classes for whole-slide image classification."""
+
+import functools
+import glob
+import os
+from typing import Callable, List, Literal, Tuple
+
+import numpy as np
+import pandas as pd
+import torch
+from torchvision.datasets import utils
+from typing_extensions import override
+
+from eva.vision.data.datasets import _validators, structs, wsi
+from eva.vision.data.datasets.classification import base
+from eva.vision.data.wsi.patching import samplers
+
+
+class PANDA(wsi.MultiWsiDataset, base.ImageClassification):
+    """Dataset class for PANDA images and corresponding targets."""
+
+    _train_split_ratio: float = 0.8
+    """Train split ratio."""
+
+    _val_split_ratio: float = 0.2
+    """Validation split ratio."""
```

Comment:
these ratios don't correspond to what we have in the readme (0.7/0.15/0.15). Also a test split is missing. (Here we have sufficient data to create a test split so we should probably create and use it)

Reply:
Good point, I wanted to discuss this with you. For panda there is also the leaderboard on kaggle, which will serve as the real test set. But still, for offline evaluation I agree that it makes sense to have train/val/test as well.

Reply:
yeah, evaluating and reporting on a val set (which was used for early stopping) is not ideal and should only be done if there is no other way (i.e. splits are given by provider / no option to split three-ways without data leakage etc.). Here the splits are created by us and there is sufficient data, so i think there is no reason to not have a test set.

Reply:
Updated this now to generate stratified train/val/test splits.